src/customRules/structuredImportsRule.ts

```diff
@@ -0,0 +1,94 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  /* tslint:disable:object-literal-sort-keys */
+  public static metadata: Lint.IRuleMetadata = {
+    'ruleName': 'structured-imports',
+    'description': 'Enforce structure to your imports. Import structure should be listed in the folloing order: modules, absolute imports,  relative parent/ancestor directories, relative sibling directors.',
+    'hasFix': false,
+    'optionsDescription': 'Not configurable.',
+    'options': null,
+    'optionExamples': null,
+    'type': 'style',
+    'typescriptOnly': false,
+  };
+  /* tslint:enable:object-literal-sort-keys */
+
+  public static STRUCTURED_IMPORTS_ABS_FIRST_ERROR = 'Imports should be listed in the following order: module imports, absolute imports, relative imports.';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const structuredImportsWalker = new StructuredImportsWalker(sourceFile, this.getOptions());
+    return this.applyWithWalker(structuredImportsWalker);
+  }
+}
+
+enum ImportType {
+  Module = 0, // 'module'
+  Absolute = 1, // '/some/folder/file'
+  RelativeAncestor = 2, // '../parentFolder'
+  RelativeSibling = 3, // './siblingFolder'
+};
+
+const importStuctureOrder = [ImportType.Module, ImportType.Absolute, ImportType.RelativeAncestor, ImportType.RelativeSibling];
+
+class StructuredImportsWalker extends Lint.RuleWalker {
+  private previousImport: ImportType;
+  private currentImport: ImportType;
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+  }
+
+  // e.g. 'import Foo from './foo';'
+  public visitImportDeclaration(node: ts.ImportDeclaration) {
+    const importPath = node.moduleSpecifier.getText();
+    if (!this.previousImport) {
+      this.previousImport = getImportType(importPath);
+      this.currentImport = getImportType(importPath);
+    } else {
+      this.currentImport = getImportType(importPath);
+      if (isCurrentImportValid(this.previousImport, this.currentImport)) {
+        this.previousImport = this.currentImport;
+      } else {
+        this.addFailure(this.createFailure(node.moduleSpecifier.getStart(), node.moduleSpecifier.getFullWidth(), Rule.STRUCTURED_IMPORTS_ABS_FIRST_ERROR));
+      }
+    }
+  }
+}
+
+function isCurrentImportValid(prevImport: ImportType, currImport: ImportType): boolean {
+  if (prevImport === currImport) {
+    return true;
+  }
+  return getNextOrderedImport(prevImport) === currImport;
+}
+
+function getNextOrderedImport(importType: ImportType) {
+  if (importType < importStuctureOrder.length - 1) {
+    return importStuctureOrder[(importType as number) + 1];
+  }
+}
+
+function getImportType(path: string): ImportType {
```

Comment:
Eh, I'd go the other way and make it uglier.  It's not going to expand beyond 4 possibilities, and most devs grok these hardcoded strings faster than pretty method names.

```ts
if (path.substr(1, 2) === './') {
  return Import.Sibling;
} else if (path.substr(1, 4) === '../') {
```

If I was going to make it prettier, I'd try [adding a static `toType(path: string)` function to the enum](https://basarat.gitbooks.io/typescript/content/docs/enums.html#enum-with-static-functions).